test/fail/M0216.mo

```diff
@@ -0,0 +1,14 @@
+func _unableToDetermineType() {
+  // Only able to infer type fields for val paths at the moment
+  let { type K } = module { type K = Nat };
+};
+
+func _requiresAnnotation() {
+  class HashMap() {
+    public type Hash = Nat64;
+  };
+  // let map : HashMap = HashMap(); // Compiles fine
+  let map = HashMap();
+  let { type Hash } = map;
```

Comment:
Ditto also unfortunate.
I wonder if this actually works when map is defined in another scope, instead of as part of the same recursive block.